src/puppetlabs/puppetdb/time.clj

```diff
@@ -235,5 +235,6 @@
   [ts]
   (tc/to-timestamp
    (if (string? ts)
-     (from-string ts)
+     (when-let [dt (from-string ts)]
+       (java.sql.Timestamp. (.getMillis dt)))
```

Comment:
My understanding is that from-string is defined to return a DateTime (theirs always does, and ours might, or might return nil).  And to-timestamp is defined to do the same, but for a Timestamp.